File: Plantilla/frontend/product-app/src/api/productService.ts
import axiosInstance from './axios';
import type { Product } from '../types/Product';

export const getProducts = async (): Promise<Product[]> => {
    const response = await axiosInstance.get('/products');
    return response.data;
};

export const createProduct = async (product: Product): Promise<Product> => {
    const response = await axiosInstance.post('/products', product);
    return response.data;
};

export const updateProduct = async (id: number, product: Product): Promise<Product> => {
    const response = await axiosInstance.put(`/products/${id}`, product);
    return response.data;
};

export const deleteProduct = async (id: number): Promise<void> => {
    await axiosInstance.delete(`/products/${id}`);
};

File: oop_fullstack_lab/DeLeon/Backend/fullstack/src/main/java/com/deleon/fullstack/service/ProductService.java
package com.deleon.fullstack.service;

import com.deleon.fullstack.entity.Product;
import com.deleon.fullstack.repository.ProductRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class ProductService {

    @Autowired
    private ProductRepository productRepository;

    // Get all products
    public List<Product> getAllProducts() {
        return productRepository.findAll();
    }

    // Get one product by ID
    public Optional<Product> getProductById(Long id) {
        return productRepository.findById(id);
    }

    // Create a new product
    public Product createProduct(Product product) {
        return productRepository.save(product);
    }

    // Update an existing product
    public Product updateProduct(Long id, Product updatedProduct) {
        return productRepository.findById(id).map(existing -> {
            existing.setName(updatedProduct.getName());
            existing.setDescription(updatedProduct.getDescription());
            existing.setPrice(updatedProduct.getPrice());
            return productRepository.save(existing);
        }).orElseThrow(() -> new RuntimeException("Product not found with id: " + id));
    }

    // Delete a product
    public void deleteProduct(Long id) {
        productRepository.deleteById(id);
    }
}
